adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -1320,23 +1337,18 @@ func replacePathParamsWithCaptureGroups(resourcePath string) string {
 }
 
 // generateSubstitutionString returns a regex that has indexes to place the path variables extracted by capture groups
-func generateSubstitutionString(endpointBasepath string, resourcePath string) string {
-	pathParaRegex := "([^/]+)"
-	pathParamIndex := 0
-	resourceRegex := replacePathParamsWithCaptureGroups(resourcePath)
-	for {
-		pathParaRemains := strings.Contains(resourceRegex, pathParaRegex)
-		if !pathParaRemains {
-			break
-		}
-		pathParamIndex++
-		resourceRegex = strings.Replace(resourceRegex, pathParaRegex, fmt.Sprintf("\\%d", pathParamIndex), 1)
+func generateSubstitutionString(resourcePath string, pathMatchType gwapiv1b1.PathMatchType) string {
```

Comment:
please remove unused methods in this file. I suppose now there should be methods which no longer useful for apk

Reply:
This is used in other files. I could remove `replacePathParamsWithCaptureGroups` since it's no longer used